plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java

```diff
@@ -106,130 +107,121 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<LoadMo
         String[] targetNodeIds = deployModelRequest.getModelNodeIds();
         // mlStats.getStat(MLNodeLevelStat.ML_NODE_EXECUTING_TASK_COUNT).increment();
         mlStats.getStat(MLNodeLevelStat.ML_NODE_TOTAL_REQUEST_COUNT).increment();
-        try {
-            DiscoveryNode[] allEligibleNodes = nodeFilter.getEligibleNodes();
-            Map<String, DiscoveryNode> nodeMapping = new HashMap();
-            for (DiscoveryNode node : allEligibleNodes) {
-                nodeMapping.put(node.getId(), node);
-            }
+        DiscoveryNode[] allEligibleNodes = nodeFilter.getEligibleNodes();
+        Map<String, DiscoveryNode> nodeMapping = new HashMap<>();
+        for (DiscoveryNode node : allEligibleNodes) {
+            nodeMapping.put(node.getId(), node);
+        }
 
-            Set<String> allEligibleNodeIds = Arrays.stream(allEligibleNodes).map(n -> n.getId()).collect(Collectors.toSet());
+        Set<String> allEligibleNodeIds = Arrays.stream(allEligibleNodes).map(DiscoveryNode::getId).collect(Collectors.toSet());
 
-            List<DiscoveryNode> eligibleNodes = new ArrayList<>();
-            List<String> nodeIds = new ArrayList<>();
-            if (targetNodeIds != null && targetNodeIds.length > 0) {
-                for (String nodeId : targetNodeIds) {
-                    if (allEligibleNodeIds.contains(nodeId)) {
-                        eligibleNodes.add(nodeMapping.get(nodeId));
-                        nodeIds.add(nodeId);
-                    }
+        List<DiscoveryNode> eligibleNodes = new ArrayList<>();
+        List<String> nodeIds = new ArrayList<>();
+        if (targetNodeIds != null && targetNodeIds.length > 0) {
+            for (String nodeId : targetNodeIds) {
+                if (allEligibleNodeIds.contains(nodeId)) {
+                    eligibleNodes.add(nodeMapping.get(nodeId));
+                    nodeIds.add(nodeId);
                 }
-            } else {
-                nodeIds.addAll(allEligibleNodeIds);
-                for (DiscoveryNode node : allEligibleNodes) {
-                    eligibleNodes.add(node);
-                }
-            }
-            if (nodeIds.size() == 0) {
-                listener.onFailure(new MLResourceNotFoundException("no eligible node found"));
-                return;
             }
+        } else {
+            nodeIds.addAll(allEligibleNodeIds);
+            eligibleNodes.addAll(Arrays.asList(allEligibleNodes));
```

Comment:
Thanks for tuning this part. Not sure if this code will perform better than the original one. I guess they are almost the same. But prefer to learn any result if you have done some experiments. Here the allEligibleNodes will be a relatively small number like 100s.

Reply:
Checked the code, `Arrays.asList` construct a new object without any iteration, `addAll` method has similar logic with `add` so there's no extra time consuming. Also did a test with 120 records, this approach has faster completion time.